### PR TITLE
Finalize generate identifiers API with improved class resolution and refinements

### PR DESCRIPTION
**Pull request by delano**
## Summary

This PR finalizes the identifier generation API improvements with several key enhancements to the Familia framework:

**Core API changes:**
- Moved from `generate_hex_id` to the cleaner `generate_id(16)` pattern
- Added `generate_lite_id` and improved `generate_trace_id` for different security needs
- Introduced `resolve_class` method for unified class resolution from Class/String/Symbol inputs

**String transformation improvements:**
- Refactored `StylizeWords` refinements to make methods testable through `StylizeWordsMethods`
- Added `demodularize` method for consistent module namespace removal
- Improved `config_name` and `familia_name` methods to handle modularized classes

**Participation relationships cleanup:**
- Replaced hash-based configuration with proper `ParticipationRelationship` data objects
- Unified collection method generation using `registered_type` lookups
- Improved method naming consistency (e.g., `participation_memberships`)

**Test infrastructure:**
- Added comprehensive test cases for new member resolution methods
- Created `TestCleanup` helper for managing anonymous test classes
- Updated secure identifier tests to match new API patterns

**Configuration updates:**
- Enhanced RuboCop configuration for parameter list limits
- Updated Talisman security scanning configuration
- Improved pre-commit hooks for CI mode

All changes maintain backward compatibility while providing cleaner, more testable interfaces for class resolution and identifier generation.

## Test plan

- [x] All existing test cases pass
- [x] New secure identifier API tests pass
- [x] String refinement tests validate demodularization
- [x] Participation relationship improvements tested
- [x] Class resolution methods tested with edge cases
- [x] Test cleanup infrastructure validated

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: lib/familia/refinements/stylize_words.rb [hunk 1]
```diff
@@ -2,104 +2,64 @@
...
+        else
+          raise ArgumentError, "Unknown stylization in first_letter: #{first_letter}"
         end
       end
     end
+
+    # Refinement that delegates to the testable methods
+    module StylizeWords
+      refine String do
+        import_methods StylizeWordsMethods
```
[the remark this answers is not in the document]
Comment (delano) [reply lib/familia/refinements/stylize_words.rb:61]: In Ruby 3.2+ it does

File: lib/familia/features/relationships/participation.rb [hunk 2]
```diff
@@ -84,21 +86,23 @@ def participates_in(target_class, collection_name, score: nil, on_destroy: :remo
             end
 
             # Store metadata for this participation relationship
-            participation_relationships << {
-              target_class: target_class,
-              target_class_name: target_class_name,
+            participation_relationships << ParticipationRelationship.new(
+              target_class: target_class,           # as passed to `participates_in`
+              target_class_name: target_class_name, # pascalized
```
[the remark this answers is not in the document]
Comment (delano) [reply lib/familia/features/relationships/participation.rb:91]: The inline comments are clarifying the expectations at that point in the code, not commenting on what is being done.
